cumulus_library/cli.py

```diff
@@ -194,166 +279,77 @@ def run_cli(args: Dict):
-            "followed by any supplied paths. Target, and all its subdirectories, "
-            "are checked for manifests. Overriding studies with the same namespace "
-            "supersede earlier ones."
-        ),
-    )
-    db.add_argument(
-        "-b",
-        "--build",
-        default=False,
-        action="store_true",
-        help=("Removes and recreates Athena tables & views for the specified studies"),
-    )
-    db.add_argument(
-        "-e",
-        "--export",
-        default=False,
-        action="store_true",
-        help=("Generates files on disk from Athena views for the specified studies"),
-    )
-    db.add_argument(
-        "-v",
-        "--verbose",
-        default=False,
-        action="store_true",
-        help=("Prints detailed SQL query info"),
-    )
-
-    aws = parser.add_argument_group("AWS config")
-    aws.add_argument("--profile", help="AWS profile", default="default")
-    aws.add_argument(
-        "--database",
-        # internally, we use PyAthena's terminology for this but the UX term is "database"
-        dest="schema_name",
-        help="Cumulus Athena database name",
-    )
-    aws.add_argument(
-        "--workgroup",
-        default="cumulus",
-        help="Cumulus Athena workgroup (default: cumulus)",
-    )
-    aws.add_argument(
-        "--region",
-        help="AWS region data of Athena instance (default: us-east-1)",
-        default="us-east-1",
-    )
-
-    return parser
-
-
 def main(cli_args=None):
     """Reads CLI input/environment variables and invokes library calls"""
     parser = get_parser()
     args = vars(parser.parse_args(cli_args))
-
-    if args["target"] is not None:
+    if args["action"] is None:
+        parser.print_usage()
+        sys.exit(1)
+    if "target" in args and args["target"] is not None:
         for target in args["target"]:
             if target == "all":
                 args["target"] = ["all"]
                 break
-    else:
-        args["target"] = ["all"]
 
-    if args["study_dir"] is not None:
+    if "study_dir" in args and args["study_dir"] is not None:
         posix_paths = []
         for path in args["study_dir"]:
             posix_paths.append(get_abs_posix_path(path))
         args["study_dir"] = posix_paths
 
-    if profile_env := os.environ.get("CUMULUS_LIBRARY_PROFILE"):
-        args["profile"] = profile_env
-    if database_env := os.environ.get("CUMULUS_LIBRARY_DATABASE"):
-        args["schema_name"] = database_env
-    if workgroup_env := os.environ.get("CUMULUS_LIBRARY_WORKGROUP"):
-        args["workgroup"] = workgroup_env
-    if region_env := os.environ.get("CUMULUS_LIBRARY_REGION"):
-        args["region"] = region_env
-    if path_dir := os.environ.get("CUMULUS_LIBRARY_PATH"):
-        args["path"] = [path_dir] + args["path"]
-
+    if "export_dir" in args and args["export_dir"] is not None:
+        args["export_dir"] = get_abs_posix_path(args["export_dir"])
+
+    arg_env_pairs = (
```

Comment:
i'm trying to mimic the original library functionality here, perhaps for ill - you've got approved pip installed libraries, and then you've got some directory where you're installing a couple in progress ones from git. if you're always putting them in the same place, you can set this once and have it apply for multiples. And then both your `build` and `export` calls can look in the same place to try and grab manifest files.

i'm 100% open to other ideas. i don't love this being an env var, but the only alternative i can think of is maintaining a db someplace that allows for registration/deregistration of studies? and that feels worse to me.

Reply:
The alternative I would suggest is to just not stress about achieving an inbox-zero command line in all possible use cases. Just make them specify it on the CLI. But if you want to support an env var, go for it.

It's just that as a matter of principle, I like to minimize user interaction points, because you ideally maintain those forever. So keeping a single CLI consistent across ten years is enough work for me, I don't love also doubling that effort but in env vars. But maybe I'm optimizing for an unrealistic use case too, and we can just break stuff more often than I default to.

Reply:
I'm going to mark this resolved based on the updates from the long convo mentioned above.